src/ActionList2/Header.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import Box from '../Box'
+import {SxProp} from '../sx'
+import {ListContext} from './List'
+
+/**
+ * Contract for props passed to the `Header` component.
+ */
+export interface HeaderProps extends React.ComponentPropsWithoutRef<'div'>, SxProp {
+  /**
+   * Style variations. Usage is discretionary.
+   *
+   * - `"filled"` - Superimposed on a background, offset from nearby content
+   * - `"subtle"` - Relatively less offset from nearby content
+   */
+  variant?: 'subtle' | 'filled'
+
+  /**
+   * Primary text which names a `Group`.
+   */
+  title?: string
+
+  /**
+   * Secondary text which provides additional information about a `Group`.
+   */
+  auxiliaryText?: string
+}
+
+/**
+ * Displays the name and description of a `Group`.
+ */
+export function Header({
+  variant = 'subtle',
+  title,
+  auxiliaryText,
+  children: _children,
```

Comment:
Why not just do
`delete props.children`?

Reply:
I think that's a no-no because props isn't a mutable object